auth/src/main/java/org/aerogear/mobile/auth/credentials/JwksManager.java

```diff
@@ -108,30 +108,35 @@ public void fetchJwks(@NonNull final KeycloakConfiguration keycloakConfiguration
         HttpResponse response = getRequest.execute();
         response.onComplete(() -> {
             JsonWebKeySet jwks = null;
-            JwksException error = null;
             // this is invoked on a background thread.
-            if (response.getStatus() == 200) {
-                String jwksContent = response.stringBody();
-                try {
-                    jwks = new JsonWebKeySet(jwksContent);
-                } catch (JoseException e) {
-                    jwks = null;
-                    error = new JwksException(e);
-                    logger.warning("failed to parse JWKS key content. content = " + jwksContent);
+            try {
+                if (response.getStatus() == 200) {
+                    String jwksContent = response.stringBody();
+                    try {
+                        jwks = new JsonWebKeySet(jwksContent);
+                    } catch (JoseException e) {
+                        logger.warning("failed to parse JWKS key content. content = "
+                            + jwksContent);
+                        if(callback != null) {
+                            callback.onError(new JwksException(e));
+                        }
+                    }
+                    if (jwks != null) {
+                        persistJwksContent(keycloakConfiguration.getRealmName(), jwksContent);
+                    }
+                } else {
+                    logger.warning("failed to fetch JWKS from server. url = " + jwksUrl
+                                    + " statusCode = " + response.getStatus());
+                    throw new JwksException("failed to fetch JWKS from server");
                 }
-                if (jwks != null) {
-                    persistJwksContent(keycloakConfiguration.getRealmName(), jwksContent);
+                if (callback != null) {
+                    if (jwks != null) {
```

Comment:
The only path to have `jwks` not null, is that it doesn't throw an exception. So if you put this in the try/catch block, you won't have to check for `jwks` not null